packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -651,4 +698,86 @@ export class IgnitionModuleBuilderImplementation<
       this.send
     );
   }
+
+  private _assertValidLibraries(
+    libraries: Record<string, ContractFuture<string>>,
+    func: (...[]: any[]) => any
+  ) {
+    for (const [libraryName, libraryFuture] of Object.entries(libraries)) {
+      if (!isContractFuture(libraryFuture)) {
+        this._throwErrorWithStackTrace(
+          `Given library '${libraryName}' is not a valid Future`,
+          func
+        );
+      }
+    }
+  }
+
+  private _assertValidValue(value: bigint | any, func: (...[]: any[]) => any) {
+    if (typeof value !== "bigint") {
+      this._throwErrorWithStackTrace(
+        `Given value option '${value}' is not a \`bigint\``,
+        func
+      );
+    }
+  }
+
+  private _assertValidFrom(
+    from: string | AccountRuntimeValue | undefined,
+    func: (...[]: any[]) => any
+  ) {
+    if (
+      !isAccountRuntimeValue(from) &&
+      typeof from !== "string" &&
+      from !== undefined
+    ) {
+      this._throwErrorWithStackTrace(
+        `Invalid type for given option "from": ${typeof from}`,
+        func
+      );
+    }
+  }
+
+  private _assertValidArtifact(
+    artifact: Artifact,
+    func: (...[]: any[]) => any
+  ) {
+    if (!isArtifactType(artifact)) {
+      this._throwErrorWithStackTrace(`Invalid artifact given`, func);
+    }
+  }
+
+  private _assertValidContract(
+    contract: ContractFuture<string>,
+    func: (...[]: any[]) => any
+  ) {
+    if (!isContractFuture(contract)) {
+      this._throwErrorWithStackTrace(`Invalid contract given`, func);
+    }
+  }
+
+  private _assertValidAddress(
+    address:
+      | string
+      | AddressResolvableFuture
+      | ModuleParameterRuntimeValue<string>,
+    func: (...[]: any[]) => any
+  ) {
+    if (
+      typeof address !== "string" &&
+      !isModuleParameterRuntimeStringValue(address) &&
```

Comment:
I think I understand your comment now. I'd say we accept module params without default here, and then validate that they actually exist and are a `string` in the second phase.